compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -514,6 +512,11 @@ class MapAllocatorCache {
     return (EntriesCount >= atomic_load_relaxed(&MaxEntriesCount));
   }
 
+  inline void advanceDecommitBase() REQUIRES(Mutex) {
+    if (DecommitBase != CachedBlock::InvalidEntry)
+      DecommitBase = Entries[DecommitBase].Prev;
+  }
+
```

Comment:
I think we can just inline this function. From its callers (now there are two), one doesn't need the validity check, another already has the check in the loop header.